federation/pkg/federation-controller/service/service_helper.go

```diff
@@ -131,9 +131,13 @@ func (cc *clusterClientCache) syncService(key, clusterName string, clusterCache
 	if isDeletion {
 		// cachedService is not reliable here as
 		// deleting cache is the last step of federation service deletion
-		_, err := fedClient.Core().Services(cachedService.lastState.Namespace).Get(cachedService.lastState.Name, metav1.GetOptions{})
+		service, err := fedClient.Core().Services(cachedService.lastState.Namespace).Get(cachedService.lastState.Name, metav1.GetOptions{})
 		// rebuild service if federation service still exists
 		if err == nil || !errors.IsNotFound(err) {
+			if err == nil && service.DeletionTimestamp != nil {
+				glog.Infof("Service %v already marked for deletion", name)
```

Comment:
Same, make this glog.V(4)? This is not an error.
Also say something like "skipping sync in underlying cluster as it has been marked for deletion"